docs/k8s/subnet.yaml

```diff
@@ -1,31 +1,43 @@
 # Copyright (c) Tailscale Inc & AUTHORS
 # SPDX-License-Identifier: BSD-3-Clause
-apiVersion: v1
-kind: Pod
+apiVersion: apps/v1
+kind: Deployment
 metadata:
   name: subnet-router
+  namespace: tailscale-system
   labels:
     app: tailscale
 spec:
-  serviceAccountName: "{{SA_NAME}}"
-  containers:
-  - name: tailscale
-    imagePullPolicy: Always
-    image: "ghcr.io/tailscale/tailscale:latest"
-    env:
-    # Store the state in a k8s secret
-    - name: TS_KUBE_SECRET
-      value: "{{TS_KUBE_SECRET}}"
-    - name: TS_USERSPACE
-      value: "true"
-    - name: TS_AUTHKEY
-      valueFrom:
-        secretKeyRef:
-          name: tailscale-auth
-          key: TS_AUTHKEY
-          optional: true
-    - name: TS_ROUTES
-      value: "{{TS_ROUTES}}"
-    securityContext:
-      runAsUser: 1000
-      runAsGroup: 1000
+  replicas: 1
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app: tailscale
+  template:
+    metadata:
+      labels:
+        app: tailscale
+    spec:
+      serviceAccountName: "{{SA_NAME}}"
+      containers:
+      - name: tailscale
+        imagePullPolicy: Always
+        image: "ghcr.io/tailscale/tailscale:latest"
+        env:
```

Comment:
We will additionally want to set `TS_AUTH_ONCE` env var to "prevent" the node logging in every time the pod is restarted, see https://github.com/tailscale/tailscale/blob/v1.56.1/cmd/k8s-operator/deploy/manifests/proxy.yaml#L31-L32